generator.rb

```diff
@@ -0,0 +1,66 @@
+# Make sure user inputted argument is long enough. If it's not, reprompt the user. Return new (or original, if already okay) input.
+def check_length(word)
+  until word.length > 1
+    print "Your word must be at least 2 characters long. > "
+    word = gets.chomp
+  end
+  return word
+end
+
+# Check if a letter is a vowel. If it is, return true. If not, return false.
+def is_vowel?(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return true
+  else
+    return false
+  end
+end
+
+# Returns an integer of how many characters there are after the first vowel in an argument.
+def segment_length(word)
+  chars_before_vowel = 0
+  word.each_char do |letter|
+    if is_vowel?(letter)
+      break
+    end
+    chars_before_vowel += 1
+  end
+  length = word.length - chars_before_vowel
+  return length
+end
```

Comment:
Yes! I'm so glad you found a way to make this logic extracted into a different method.